src/context/box-repository.ts

```diff
@@ -93,6 +94,15 @@ export class BoxRepository extends BaseRepository<BoxEntity> {
         query = query.andWhere(`${this.alias}.settlement_height <= :maxHeight`, { maxHeight });
       }
 
+      if (options.where?.boxId && boxIds) {
+        boxIds.push(options.where.boxId);
+        delete options.where.boxId;
+      }
+
+      if (boxIds) {
```

Comment:
If an empty array is submitted, it will result in a broken query, please add a check for it.

Reply:
Fixed.